debezium-core/src/main/java/io/debezium/jdbc/JdbcConnection.java

```diff
@@ -1710,29 +1710,30 @@ public String getQualifiedTableName(TableId tableId) {
         return tableId.schema() + "." + tableId.table();
     }
 
-    public Map<String, Object> reselectColumns(Table table, List<String> columns, List<String> keyColumns, List<Object> keyValues, Struct source)
+    public boolean reselectColumns(Table table, List<String> columns, List<String> keyColumns, List<Object> keyValues, Struct source,
+                                   ParameterResultSetConsumer resultConsumer)
             throws SQLException {
         final String query = String.format("SELECT %s FROM %s WHERE %s",
                 columns.stream().map(this::quoteIdentifier).collect(Collectors.joining(",")),
                 quotedTableIdString(table.id()),
                 keyColumns.stream().map(key -> key + "=?").collect(Collectors.joining(" AND ")));
-        return reselectColumns(query, table.id(), columns, keyValues);
+        return reselectColumns(query, table.id(), columns, keyValues, resultConsumer);
     }
 
-    protected Map<String, Object> reselectColumns(String query, TableId tableId, List<String> columns, List<Object> bindValues) throws SQLException {
-        final Map<String, Object> results = new HashMap<>();
+    protected boolean reselectColumns(String query, TableId tableId, List<String> columns, List<Object> bindValues, ParameterResultSetConsumer resultConsumer)
+            throws SQLException {
+        final boolean[] flags = { false };
         prepareQuery(query, bindValues, (params, rs) -> {
```

Comment:
Could we use `prepareQueryAndMap` and have the lambda return true/false to the caller?

Reply:
Done in ec5dde6916c2894b75ab8ec4ebd308da9121df15

Reply:
There are also `io.debezium.jdbc.JdbcConnection.singleResultMapper(ResultSetExtractor<T>, String)` and `io.debezium.jdbc.JdbcConnection.querySingleValue(Connection, String, StatementPreparer, ResultSetExtractor<T>)`. Maybe it would be good to provide alternatives that emit a warning instead of exception and we'll get best of the both worlds.

Reply:
Done without helper methods in 1a37cc646945ac72903d68a19e1a748b808b71b2.
@jpechane, I looked at the methods you suggested but sorry I didn't get what version of them could make reselectColumns cleaner. It looks like it's sort of special case where we don't need a mapper or an extractor, but a consumer plus an indication whether it was called or not.
For now I think it's better to keep it as is. But if you give me a hint I'll look at it again)

Reply:
Well, the methods I pointed do have exactly the same structure as your plain JDBC code. So I propose to introduce generic methods that are the same as the ones I pointed with single difference, instead of throwing an error they'll log warning - exatcly as you do and use them here. The general methods can be useful in the codebase elsewhere.

Reply:
@jpechane , please look at fee7cbbe85c397560714d6160243922f1543e2a4, I hope it's close to how you see it. Additionally I've introduced `statementValuesSetter` method that accept a list of statement values and returns `StatementPreparer`. Let me know if you want me to remove it.